akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EffectImpl.scala

```diff
@@ -3,37 +3,30 @@
  */
 package akka.persistence.typed.internal
 
-import akka.persistence.typed.{ javadsl ⇒ j }
+import akka.persistence.typed.javadsl
+import akka.persistence.typed.scaladsl
 
 import scala.collection.{ immutable ⇒ im }
-import akka.annotation.InternalApi
-import akka.persistence.typed.scaladsl.PersistentBehaviors.{ ChainableEffect, Effect }
+import akka.annotation.{ DoNotInherit, InternalApi }
 
-/**
- * INTERNAL API
- */
+/** INTERNAL API */
 @InternalApi
-private[akka] abstract class EffectImpl[+Event, State] extends j.Effect[Event, State] with Effect[Event, State] {
+private[akka] abstract class EffectImpl[+Event, State] extends javadsl.Effect[Event, State] with scaladsl.Effect[Event, State] {
   /* All events that will be persisted in this effect */
   override def events: im.Seq[Event] = Nil
 
   /* All side effects that will be performed in this effect */
   override def sideEffects[E >: Event]: im.Seq[ChainableEffect[E, State]] = Nil
 }
 
-/**
- * INTERNAL API
- */
+/** INTERNAL API */
 @InternalApi
 private[akka] object CompositeEffect {
```

Comment:
Maybe put all these into the companion object of `EffectImpl` then a single internal api marker is enough.